Library/Homebrew/service.rb

```diff
@@ -410,12 +419,14 @@ def to_plist
 
       if @sockets.present?
         base[:Sockets] = {}
-        base[:Sockets][:Listeners] = {
-          SockNodeName:    @sockets[:host],
-          SockServiceName: @sockets[:port],
-          SockProtocol:    @sockets[:type].upcase,
-          SockFamily:      "IPv4v6",
-        }
+        @sockets.each do |name, info|
+          base[:Sockets][name] = {
+            SockNodeName:    info[:host],
+            SockServiceName: info[:port],
+            SockProtocol:    info[:type].upcase,
+            SockFamily:      "IPv4v6",
```

Comment:
Would it be possible to allow customizing this value too? I don't need IPv6 for my formula

Reply:
Actually my [app](https://github.com/owenthereal/candy) doesn't work with `IPv6`

Reply:
I think it was just chosen so that the sockets are generic and work with either ipv4 and/or ipv6 automatically.

From the man page (`man launchd.plist`):

> SockFamily <string>
>
> This optional key can be used to specifically request that "IPv4" or "IPv6" socket(s) be created. An additional option, "IPv4v6" indicates that a single socket that listens for both IPv4 and IPv6 connections should be created.

Reply:
I ran into an issue setting it to `IPv4v6` on mac 14.0: https://github.com/bored-engineer/go-launchd/issues/2#issuecomment-1742321770. If I set it to `IPv4`, things work, but not `IPv4v6` or `IPv6`. It doesn't appear that the socket works with ipv4 & ipv6 automatically. Users should have the flexibility to choose `SockFamily` IMO.

Reply:
If `IPv4v6` is the hard-coded value, how would one define `SockNodeName` so the address works with ipv4 & ipv6? 

Ref: https://github.com/bored-engineer/go-launchd/issues/2#issuecomment-1742326698

Reply:
@SMillerDev Do you remember why `IPv4v6` was chosen as the default value for `SockFamily` here?

@owenthereal I see that the old version of your formula didn't even specify `SockFamily` at all. It's not obvious from reading the docs what the default is for that field. Maybe it defaults to ipv4.

Reply:
I think the easiest thing here would be to just check if the port is using an ipv4 or ipv6 address and just set it automatically instead of using ipv4v6. The few use cases for this in core and on Github all seem to use ipv4 anyway.

Reply:
I'm also surprised this hasn't come up before with the formulas that are using sockets since they all use ipv4 already.

Reply:
I think making it a configurable option with ipv4 default makes the most sense with the information I have now.

Reply:
I'd rather not add another option because I'm lazy and that would require more work on my part. It would require coordinating the update to the three packages that currently use sockets with the API JSON and a new brew release. Hopefully, just figuring it out from the host is good enough for most uses cases.

Reply:
Actually, I overlooked that we don't even allow ipv6 addresses right now anyway because of a bug in input validation. We could probably just default to ipv4 since nobody seems to be clamoring for ipv6 sockets. That would be the simplest option.

Reply:
Or we could just remove the `SockFamily` parameter entirely. Two of the three formulae didn't have it specified in the plist before https://github.com/Homebrew/homebrew-core/pull/93612/ and the last one is `bitlbee` which should work with the default as well.